### ISSUE TITLE
Haskell not included in BINDINGS=ALL

### ISSUE THREAD

**Issue by ingwinlu** [ElektraInitiative/libelektra#1890]
## Steps to Reproduce the Problem
```sh
cmake -DENABLE_COVERAGE=ON -DPLUGINS=ALL -DBINDINGS=ALL $WORKSPACE
Exclude Plugin haskell because haskell bindings are not included in the cmake configuration
```

## Expected Result
Haskell should be included in `BINDINGS=ALL`

## Actual Result
Haskell is not included in ALL

## System Information

- Elektra Version: master

## Workaround
```sh
cmake -DENABLE_COVERAGE=ON -DPLUGINS=ALL -DBINDINGS="ALL;haskell" $WORKSPACE
```

**Comment by markus2330** [comment]
Thank you for reporting!

Can you please also post the part why the haskell binding is excluded?

**Comment by ingwinlu** [comment]
> Can you please also post the part why the haskell binding is excluded?

It does not say, but from what I gathered it is just because it is not included in the BINDING=ALL target

**Comment by markus2330** [comment]
@waht Can you fix that bindings print the reason for why they got excluded?

And why is the haskell binding excluded in this case? BINDING=ALL should also include experimental bindings, shouldn't it?

**Comment by ingwinlu** [comment]
After having a fresh look at this I think it is actually working as expected.
Experimental should not be included in ALL.

**Comment by waht** [comment]
No, ALL should include everything - I created PR #1894 to fix this.

There were two things wrong which I had changed but it seems like they were lost at some point (e.g. when merging):
- The haskell binding had no call to `add_binding ()`  - this is the reason why there was no "Include Binding ..." message.
- The LibAddHaskellPlugin module did use the `BINDINGS` variable directly (like we had before having categories for bindings)
